commands/payment_channel_daemon_test.go

```diff
@@ -1,547 +1,620 @@
 package commands_test
 
 import (
-	"fmt"
+	"context"
+	"io/ioutil"
+	"math/big"
+	"os"
 	"strings"
-	"sync"
 	"testing"
-
-	"gx/ipfs/QmR8BauakNcBa3RbE4nbQu76PDiJgoQgz8AJdhJuiU4TAw/go-cid"
-	cbor "gx/ipfs/QmcZLyosDwMKdB6NLRsiss9HXzDPhVhhRtPy67JFKTDQDX/go-ipld-cbor"
+	"time"
 
 	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/assert"
 	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/require"
-
+	"gx/ipfs/QmQmhotPUzVrMEWNK3x1R5jQ5ZHWyL7tVUrmRPjrBrvyCb/go-ipfs-files"
+	cbor "gx/ipfs/QmcZLyosDwMKdB6NLRsiss9HXzDPhVhhRtPy67JFKTDQDX/go-ipld-cbor"
 	"gx/ipfs/QmekxXDhCxCJRNuzmHreuaT3BsuJcsjcXWNrtV9C8DRHtd/go-multibase"
 
 	"github.com/filecoin-project/go-filecoin/actor/builtin/paymentbroker"
 	"github.com/filecoin-project/go-filecoin/address"
-	"github.com/filecoin-project/go-filecoin/fixtures"
 	th "github.com/filecoin-project/go-filecoin/testhelpers"
+	"github.com/filecoin-project/go-filecoin/tools/fast"
+	"github.com/filecoin-project/go-filecoin/tools/fast/series"
+	localplugin "github.com/filecoin-project/go-filecoin/tools/iptb-plugins/filecoin/local"
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
-func TestPaymentChannelCreateSuccess(t *testing.T) {
-	t.Parallel()
-	assert := assert.New(t)
-
-	d := makeTestDaemonWithMinerAndStart(t)
-	defer d.ShutdownSuccess()
-
-	args := []string{"paych", "create"}
-	args = append(args, "--from", fixtures.TestAddresses[0], "--gas-price", "0", "--gas-limit", "300")
-	args = append(args, fixtures.TestAddresses[1], "10000", "20")
+func init() {
+	_, _ = os.Getwd()
 
-	paymentChannelCmd := d.RunSuccess(args...)
-	messageCid, err := cid.Parse(strings.Trim(paymentChannelCmd.ReadStdout(), "\n"))
-	require.NoError(t, err)
-
-	var wg sync.WaitGroup
+	series.GlobalSleepDelay = time.Second * 5
+}
 
-	wg.Add(1)
-	go func() {
-		wait := d.RunSuccess("message", "wait",
-			"--return",
-			"--message=false",
-			"--receipt=false",
-			messageCid.String(),
-		)
-		_, ok := types.NewChannelIDFromString(strings.Trim(wait.ReadStdout(), "\n"), 10)
-		assert.True(ok)
-		wg.Done()
-	}()
+func VoucherFromString(data string) (paymentbroker.PaymentVoucher, error) {
```

Comment:
This should probably be a function in `types`? However, `PaymentVoucher` is not defined there. Should we move it there?

Reply:
Yes, they do appear to be the same exact code

Reply:
Opened an issue to move payment vouchers: https://github.com/filecoin-project/go-filecoin/issues/2527